cipd_packages/codesign/lib/src/file_codesign_visitor.dart

```diff
@@ -255,6 +255,33 @@ update these file paths accordingly.
       }
       log.info('Child file of directory ${directory.basename} is ${entity.basename}');
     }
+    final String directoryExtension = directory.basename.split('.').last;
+    if (directoryExtension == 'framework' || directoryExtension == 'xcframework') {
+      final List<String> args = <String>[
+        '/usr/bin/codesign',
+        '--keychain',
+        'build.keychain', // specify the keychain to look for cert
+        '-f', // force. Possible re-signing of FlutterMacOS.framework, Flutter.xcframework/ios-arm64/Flutter.framework etc.
```

Comment:
Does it re-sign them? Should it be using `--options=runtime` then (just needed for macOS)?  In which case maybe this the actual `codesign` args logic should be shared?
https://github.com/flutter/cocoon/blob/58930df6c1d291136bcb59a7a94532520eeb24f6/cipd_packages/codesign/lib/src/file_codesign_visitor.dart#L326C8-L326C25

Reply:
Good point! looked up `--options=runtime` and it seems to protect runtime integrity. Will add this flag soon.

Sorry for the confusion, by potential re-signing I was referring to:
if the major executable of a bundle is already signed, then when we later code sign the bundle itself the signature of bundle might already exist.
e.g. if we code sign in an inside-out fashion, the main executable `FlutterMacOS.framework/Versions/A/FlutterMacOS` will be codesigned before we codesign `FlutterMacOS.framework`. code signing the latter might fail if a signature from the former already exists, but if we use the `--force` flag it will replace the existing signature if it exists.
I will try to make this comment clearer or delete it.

and sure I will look into refactoring the code

Reply:
refactored and updated